Mathlib/GroupTheory/Perm/Support.lean

```diff
@@ -166,111 +146,188 @@ theorem Disjoint.mul_apply_eq_iff {σ τ : Perm α} (hστ : Disjoint σ τ) {a
 theorem Disjoint.mul_eq_one_iff {σ τ : Perm α} (hστ : Disjoint σ τ) :
     σ * τ = 1 ↔ σ = 1 ∧ τ = 1 := by simp_rw [ext_iff, one_apply, hστ.mul_apply_eq_iff, forall_and]
 
-theorem Disjoint.zpow_disjoint_zpow {σ τ : Perm α} (hστ : Disjoint σ τ) (m n : ℤ) :
-    Disjoint (σ ^ m) (τ ^ n) := fun x =>
-  Or.imp (fun h => zpow_apply_eq_self_of_apply_eq_self h m)
-    (fun h => zpow_apply_eq_self_of_apply_eq_self h n) (hστ x)
-
-theorem Disjoint.pow_disjoint_pow {σ τ : Perm α} (hστ : Disjoint σ τ) (m n : ℕ) :
-    Disjoint (σ ^ m) (τ ^ n) :=
-  hστ.zpow_disjoint_zpow m n
-
 end Disjoint
 
 section IsSwap
 
-variable [DecidableEq α]
+/-- `f.IsSwapOn x y` indicates that the permutation `f` is a transposition of the two elements `x`
+and `y`. Note that this means that `f` might be trivial, if `x = y`. -/
+def IsSwapOn (f : Perm α) (x y : α) : Prop :=
+  (∀ {z}, z ≠ x → z ≠ y → f z = z) ∧ f x = y ∧ f y = x
```

Comment:
Can you make this a separate PR?

Reply:
Will do - there was a good reason I did it here I think but I can't remember what it was and I can surely disentangle it.